### PR TITLE
feat: add microsoft teams provider

### PR DESCRIPTION
**Pull request by danilovaz**
Closes # 

## 📑 Description

Add Microsoft Teams provider

## ✅ Checks

- [X] My pull request adheres to the code style of this project
- [X] My code requires changes to the documentation
- [X] I have updated the documentation as required
- [X] All the tests have passed

## ℹ Additional Information

I'm not fluent in python, so I apologize in advance if something is too bizarre. I appreciate any tips to improve my PR

### CODE REVIEW

**Review by shahargl — COMMENTED** [review]
@danilovaz all in all its great! would like to suggest a few additions to the docs so it'll be easier to integrate with this provider.
out of curiosity - what are your use cases for Keep? are you going to use it?

File: docs/docs/providers/documentation/teams.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+:::note Brief Description
+Teams Provider is a provider that allows to notify alerts to Microsoft Teams chats.
+:::
+
+## Inputs
+The `notify` function in the `TeamsProvider` class takes the following parameters:
+```python
+kwargs (dict):
+    message (str): The message to send. *Required*
+    typeCard (str): The card type. (MessageCard is default)
```
[the remark this answers is not in the document]
Comment (danilovaz) [reply docs/docs/providers/documentation/teams.md:16]: sur, I'll work on some better examples. the Teams Notifications API makes N scenarios really possible

[the remark this answers is not in the document]
Comment (shahargl) [reply docs/docs/providers/documentation/teams.md:16]: @danilovaz 
Super cool. so add the docs and I'll merge it.
Let me know how I can help make Keep usable for your use cases.

As part of our learning process, we try to find interesting use cases for Keep to make it more robust. 
My email is shahar@keephq.dev feel free to reach out (I'll be super happy to talk with you :))

**Comment by talboren** [comment]
@danilovaz any progress with this one? :)

**Comment by danilovaz** [comment]
> @danilovaz any progress with this one? :)

hey @talboren I haven't been able to get around to it yet. a lot of work these past few days. I will be able to finish my reasoning in this PR throughout the week

**Comment by talboren** [comment]
> > @danilovaz any progress with this one? :)
> 
> hey @talboren I haven't been able to get around to it yet. a lot of work these past few days. I will be able to finish my reasoning in this PR throughout the week

Amazing @danilovaz, looking forward

**Comment by talboren** [comment]
@danilovaz still waiting for your contribution on this one, if you'd like me to finish it just lmk! 🙏🏼

File: docs/docs/providers/documentation/teams.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (talboren) [reply docs/docs/providers/documentation/teams.md:16]: Created issue https://github.com/keephq/keep/issues/111 for that, let's merge it in the meanwhile

**Pull request merged**